sql/parser/eval.go

```diff
@@ -401,14 +401,33 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (DBool, error){
 
 	cmpArgs{Like, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
 		pattern := regexp.QuoteMeta(string(right.(DString)))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Replace LIKE specific wildcards with standard wildcards
 		pattern = strings.Replace(pattern, "%", ".*", -1)
 		pattern = strings.Replace(pattern, "_", ".", -1)
-		pattern = fmt.Sprintf("^%s$", pattern)
 
 		// TODO(pmattis): Can we cache this regexp compilation?
 		re := regexp.MustCompile(pattern)
 		return DBool(re.MatchString(string(left.(DString)))), nil
 	},
+
+	cmpArgs{SimilarTo, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
+		pattern := string(right.(DString))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Escape all periods, because they are NOT metacharacters for SQL-style regexp
+		pattern = strings.Replace(pattern, ".", "\\.", -1)
+		// Replace SQL-style regexp specific wildcards with standard regexp wildcards
+		pattern = strings.Replace(pattern, "%", ".*", -1)
+		pattern = strings.Replace(pattern, "_", ".", -1)
```

Comment:
I don't think simple string replacement is sufficient. Consider the regex `[_%]*`. The above would replace that with `[..*]*` which doesn't seem correct. I think it is worthwhile to move this regex syntax transformation into a function which can be independently unit-tested.